hera_pspec/pspecdata.py

```diff
@@ -1085,11 +1094,14 @@ def pspec(self, bls1, bls2, dsets, input_data_weight='identity', norm='I', taper
                     pol_data.extend(pv.T)
 
                     # insert integration info
-                    wgts1 = (~dset1.get_flags(bl1)).astype(np.float)
-                    nsamp1 = np.sum(dset1.get_nsamples(bl1) * wgts1, axis=1) / np.sum(wgts1, axis=1).clip(1, np.inf)
-                    wgts2 = (~dset2.get_flags(bl2)).astype(np.float)
-                    nsamp2 = np.sum(dset2.get_nsamples(bl2) * wgts2, axis=1) / np.sum(wgts2, axis=1).clip(1, np.inf)
-                    pol_ints.extend(np.mean([nsamp1, nsamp2], axis=0))
+                    wgts1 = self.w(key1).T
+                    nsamp1 = np.sum(dset1.get_nsamples(bl1)[:, self.spw_range[0]:self.spw_range[1]] * wgts1, axis=1) / np.sum(wgts1, axis=1).clip(1, np.inf)
```

Comment:
Please add a comment to make it clear what's being done here (it seems to be ignoring NaNs?).

Reply:
sure. the `clip()` is just ensuring that there is no zerodivision error for times that are entirely flagged across frequency